pylint/checkers/variables.py

```diff
@@ -603,10 +612,31 @@ def get_next_to_consume(self, node):
             found_nodes = [
                 n
                 for n in found_nodes
-                if not isinstance(n.statement(), nodes.ExceptHandler)
-                or n.statement().parent_of(node)
+                if not isinstance(n.statement(future=True), nodes.ExceptHandler)
+                or n.statement(future=True).parent_of(node)
             ]
 
+        # Filter out assignments in an Except clause that node is not
+        # contained in, assuming they may fail
+        if found_nodes:
+            filtered_nodes = [
+                n
+                for n in found_nodes
+                if not (
+                    isinstance(n.statement(future=True).parent, nodes.ExceptHandler)
+                    and isinstance(
+                        n.statement(future=True).parent.parent, nodes.TryExcept
+                    )
+                )
+                or n.statement(future=True).parent.parent_of(node)
+            ]
+            difference = [n for n in found_nodes if n not in filtered_nodes]
+            try:
+                self.consumed_uncertain[node.name] += difference
+            except KeyError:
+                self.consumed_uncertain[node.name] = difference
```

Comment:
Does a `defaultdict` make sense here? I know we're not using it in the other attributes, but that shouldn't hold us back from using better practices in new code.